### PR TITLE
finish count

### PR DESCRIPTION
**Pull request by JackTan25**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/sql/colexec/agg/count/count.go [hunk 1]
```diff
@@ -145,3 +147,222 @@ func (r *Count) Eval(zs []int64, _ *mheap.Mheap) *vector.Vector {
...
+	return &DistCount{
+		Ityp: []types.Type{ityp},
+		Otyp: otyp,
+	}
+}
+func CRC32(input string) uint32 {
+	bytes := []byte(input)
+	return crc32.ChecksumIEEE(bytes)
+}
+func getValue(vec *vector.Vector, sel int64) uint32 {
```
Comment (nnsgmsone) [inline pkg/sql/colexec/agg/count/count.go:165]: Why convert to a string, isn't that a poor performance?

File: pkg/sql/colexec/agg/count/count.go [hunk 2]
```diff
@@ -145,3 +147,222 @@ func (r *Count) Eval(zs []int64, _ *mheap.Mheap) *vector.Vector {
...
+		r.Add(i, int64(j), int64(z), vec)
+	}
+}
+
+// r[x] += a[y]
+// the old implementation in ring is strange and obviously wrong.
+func (r *DistCount) Merge(a agg.Agg[any], x, y int64) {
+	ar := a.(*DistCount)
+	for k := range ar.Ms[y] {
+		z, ok := r.Ms[x][k]
```
Comment (nnsgmsone) [inline pkg/sql/colexec/agg/count/count.go:343]: The logic here is wrong

File: pkg/sql/colexec/agg/count/count.go [hunk 3]
```diff
@@ -145,3 +147,222 @@ func (r *Count) Eval(zs []int64, _ *mheap.Mheap) *vector.Vector {
 		Nsp:  &nulls.Nulls{},
 	}
 }
+
+//for distinct count, the arguments can be many, So I make a simple way to splve it (need to optimize)
+//like count(distinct a,b,c),I will parse a,b,c as string type to get stra,strb,strc. Then I will use
+//Crc32 to hash them and I get uint32 of a,b,c, they are u32a,u32b,u32c. then add them all u32a+u32b+u32c
+//so I get the hashvalue of multicols. But I don't know whether its effect is good.(todo: optimize it)
```
Comment (fengttt) [inline pkg/sql/colexec/agg/count/count.go:154]: Wait -- are you using Crc32 to count distinct?    This won't work.

**Review by fengttt — CHANGES_REQUESTED** [review]

File: pkg/sql/colexec/agg/count/count.go (hunk 3 above)
Comment (nnsgmsone) [reply pkg/sql/colexec/agg/count/count.go:154]: Yes, this may conflict

Comment (JackTan25) [reply pkg/sql/colexec/agg/count/count.go:154]: Yeah, I have modified. Before, I think of the count(distinct a,b,c), and I found a way: that's up. But now I know it's bad. Now, we just think about the only one columns. If we think about multi columns. I will refer to the group.go

Comment (fengttt) [reply pkg/sql/colexec/agg/count/count.go:154]: Even one column won't work.   CRC is simply not good enough.    Why not just count distinct tuples?

Comment (nnsgmsone) [reply pkg/sql/colexec/agg/count/count.go:154]: I have talk him to refer to the logic of join, make a key out of each column and insert it into the hashtable.

**Pull request closed without merging**